utils/index.ts

```diff
@@ -9,17 +9,21 @@ import { Event } from "../pages/api/types";
  * We decided to use a greedy algorithm to group events.
  * This is a simple algorithm that should produce optimal solution for scheduling problems.
  */
-export const getGroupedEvents = (eventsData: Event[]) => {
-  let events = [...eventsData];
-  const groups: Event[][] = [];
+export const getGroupedEvents = (eventsData: CalendarEvent[]) => {
+  const sortedByTimeEvents = eventsData.sort(
+    (a, b) => a.startTime - b.startTime
+  );
```

Comment:
This looks like an issue, `sort` sorts the data in place. Should you mutate the data passed as a parameter?

Reply:
You're right. Fixed!